### PR TITLE
route_and_fill proptests

### PR DESCRIPTION
**Pull request by zbuc**
## Describe your changes

This adds a proptest function for exercising `route_and_fill`. In particular, it checks for the issue fixed in https://github.com/penumbra-zone/penumbra/pull/4283 . After backporting the proptest to the commit previous to #4283, I was able to reproduce the described issue (check out the `4299_route_and_fill_proptest_pre_4283` branch and `cargo test --package penumbra-dex --lib -- component::router::tests::test_paths_dont_lose_money` to reproduce) and create a failing proptest definition. I then brought this failing proptest definition over to the main branch, ran the proptests, and validated that the test now passes.

It makes sense that the issue fixed in #4283 could have caused the behavior described in #4299.

This proptest introduces two checks: that value is never lost during a trace (this may happen due to rounding, and this test _may actually be undesirable_) and that the same position is never executed against twice for a given swap execution (is this desirable? It seems possible for there to be cases where a position could be executed against in multiple traces).

## Issue ticket number and link

Closes https://github.com/penumbra-zone/penumbra/issues/4299

## Checklist before requesting a review

- [x] If this code contains consensus-breaking changes, I have added the "consensus-breaking" label. Otherwise, I declare my belief that there are not consensus-breaking changes, for the following reason:

  > only tests

### CODE REVIEW

File: crates/core/component/dex/proptest-regressions/component/router/tests.txt [hunk 1]
```diff
@@ -0,0 +1,8 @@
...
+cc 3854e480dc5fe1428eecc71207e8ffbf2859539ef407c149b163ed19703e212b # shrinks to input_amount = 1625583888576709026, num_paths_1 = 15, num_paths_2 = 5, num_paths_3 = 6
```
[the remark this answers is not in the document]
Comment (zbuc) [reply crates/core/component/dex/proptest-regressions/component/router/tests.txt:8]: grr, I think I committed the wrong failing proptest, the current strategy shouldn't produce those values... 1s

[the remark this answers is not in the document]
Comment (zbuc) [reply crates/core/component/dex/proptest-regressions/component/router/tests.txt]: @erwanor i just pushed up a new proptest file, it had some junk in it from previous development runs

[the remark this answers is not in the document]
Comment (zbuc) [reply crates/core/component/dex/proptest-regressions/component/router/tests.txt]: (new one fails on `_pre_4283` branch, passes on this one)

[the remark this answers is not in the document]
Comment (erwanor) [reply crates/core/component/dex/proptest-regressions/component/router/tests.txt]: sick thanks, taking a look!

**Comment by zbuc** [comment]
@erwanor I think this PR doesn't fulfill any current testing requirements. Are there any tests you can think of that would be more appropriate, or should we just close this out?

**Comment by erwanor** [comment]
Yeah I think we can close this and revisit in a few weeks.

**Pull request closed without merging**